<launch>
  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector_0" output="screen">
    <!-- Remap topic required by the node to custom topics -->
    <remap from="image_rect" to="/usb_cam_0/image_rect" />
    <!--remap from="image_rect_1" to="/usb_cam_1/image_rect" /-->
    <!-- remap from="image_rect_2" to="/usb_cam_2/image_rect" /-->
    <remap from="camera_info" to="/usb_cam/camera_info" />
    <remap from="tag_detections" to="/tag_detections_0" />

    <!-- Optional: Subscribe to the compressed stream-->
    <param name="image_transport" type="str" value="compressed" />

    <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
    <param name="tag_family" type="str" value="36h11" />

    <!-- Enable projected optical measurements for more accurate tag transformations -->
    <!-- This exists for backwards compatability and should be left true for new setups -->
    <param name="projected_optics" type="bool" value="true" />

    <!-- Describe the tags -->
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.166}, 
      {id: 1, size: 0.166},
      {id: 2, size: 0.166},
      {id: 3, size: 0.166},
      {id: 4, size: 0.166},
      {id: 5, size: 0.166},
      {id: 6, size: 0.166},
      {id: 7, size: 0.166},
      {id: 8, size: 0.166},
      {id: 9, size: 0.166},
      {id: 10, size: 0.166}]
    </rosparam>
  </node>
  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector_1" output="screen">
    <!-- Remap topic required by the node to custom topics -->
    <remap from="image_rect" to="/usb_cam_1/image_rect" />
    <remap from="camera_info" to="/usb_cam/camera_info" />
    <remap from="tag_detections" to="/tag_detections_1" />
    <!--remap from="localization_data" to="/localization_unused" /-->

    <!-- Optional: Subscribe to the compressed stream-->
    <param name="image_transport" type="str" value="compressed" />

    <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
    <param name="tag_family" type="str" value="36h11" />

    <!-- Enable projected optical measurements for more accurate tag transformations -->
    <!-- This exists for backwards compatability and should be left true for new setups -->
    <param name="projected_optics" type="bool" value="true" />

    <!-- Describe the tags -->
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.166}, 
      {id: 1, size: 0.166},
      {id: 2, size: 0.166},
      {id: 3, size: 0.166},
      {id: 4, size: 0.166},
      {id: 5, size: 0.166},
      {id: 6, size: 0.166},
      {id: 7, size: 0.166},
      {id: 8, size: 0.166},
      {id: 9, size: 0.166},
      {id: 10, size: 0.166}]
    </rosparam>
  </node>
  <node pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector_2" output="screen">
    <!-- Remap topic required by the node to custom topics -->
    <remap from="image_rect" to="/usb_cam_2/image_rect" />
    <remap from="camera_info" to="/usb_cam/camera_info" />
    <remap from="tag_detections" to="/tag_detections_2" />
    <!--remap from="localization_data" to="/localization_unused" /-->

    <!-- Optional: Subscribe to the compressed stream-->
    <param name="image_transport" type="str" value="compressed" />

    <!-- Select the tag family: 16h5, 25h7, 25h9, 36h9, or 36h11(default) -->
    <param name="tag_family" type="str" value="36h11" />

    <!-- Enable projected optical measurements for more accurate tag transformations -->
    <!-- This exists for backwards compatability and should be left true for new setups -->
    <param name="projected_optics" type="bool" value="true" />

    <!-- Describe the tags -->
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.166}, 
      {id: 1, size: 0.166},
      {id: 2, size: 0.166},
      {id: 3, size: 0.166},
      {id: 4, size: 0.166},
      {id: 5, size: 0.166},
      {id: 6, size: 0.166},
      {id: 7, size: 0.166},
      {id: 8, size: 0.166},
      {id: 9, size: 0.166},
      {id: 10, size: 0.166}]
    </rosparam>
  </node>
</launch>
